src/CommunityToolkit.Maui/Behaviors/PlatformBehaviors/Touch/GestureManager.cs

```diff
@@ -0,0 +1,828 @@
+		{
+			button.SendClicked();
+		}
+
+		sender.RaiseCompleted();
+	}
+
+	static void HandleCollectionViewSelection(TouchBehavior sender)
+	{
+		if (sender.Element is null 
+			|| !TryFindParentElementWithParentOfType(sender.Element, out var child, out CollectionView? collectionView))
+		{
+			return;
+		}
+
+		var selectedItem = child.BindingContext;
+
+		switch (collectionView.SelectionMode)
+		{
+			case SelectionMode.Single:
+				collectionView.SelectedItem = selectedItem;
+				break;
+
+			case SelectionMode.Multiple:
+				var selectedItems = collectionView.SelectedItems ?? [];
+
+				if (!selectedItems.Remove(selectedItem))
+				{
+					selectedItems.Add(selectedItem);
+				}
+
+				collectionView.UpdateSelectedItems(selectedItems);
+				break;
+
+			case SelectionMode.None:
+				break;
+
+			default:
+				throw new NotSupportedException($"{nameof(SelectionMode)} {collectionView.SelectionMode} is not yet supported");
+		}
+		
+		static bool TryFindParentElementWithParentOfType<T>(in VisualElement element, [NotNullWhen(true)] out VisualElement? child, [NotNullWhen(true)] out T? parent) where T : VisualElement
+		{
+			ArgumentNullException.ThrowIfNull(element);
+
+			VisualElement? searchingElement = element;
+		
+			child = null;
+			parent = null;
+
+			while (searchingElement?.Parent is not null)
+			{
+				if (searchingElement.Parent is not T parentElement)
+				{
+					searchingElement = searchingElement.Parent as VisualElement;
+					continue;
+				}
+
+				child = searchingElement;
+				parent = parentElement;
+
+				return true;
+			}
+
+			return false;
+		}
+	}
+
+	static async Task SetBackgroundImage(TouchBehavior sender, TouchState touchState, HoverState hoverState, TimeSpan duration, CancellationToken token)
+	{
+		var normalBackgroundImageSource = sender.NormalBackgroundImageSource;
+		var pressedBackgroundImageSource = sender.PressedBackgroundImageSource;
+		var hoveredBackgroundImageSource = sender.HoveredBackgroundImageSource;
+
+		var aspect = sender.BackgroundImageAspect;
+		var source = normalBackgroundImageSource;
+		if (touchState is TouchState.Pressed)
+		{
+			if (sender.IsSet(TouchBehavior.PressedBackgroundImageAspectProperty))
+			{
+				aspect = sender.PressedBackgroundImageAspect;
+			}
+
+			source = pressedBackgroundImageSource;
+		}
+		else if (hoverState is HoverState.Hovered)
+		{
+			if (sender.IsSet(TouchBehavior.HoveredBackgroundImageAspectProperty))
+			{
+				aspect = sender.HoveredBackgroundImageAspect;
+			}
+
+			if (sender.IsSet(TouchBehavior.HoveredBackgroundImageSourceProperty))
+			{
+				source = hoveredBackgroundImageSource;
+			}
+		}
+		else
+		{
+			if (sender.IsSet(TouchBehavior.NormalBackgroundImageAspectProperty))
+			{
+				aspect = sender.NormalBackgroundImageAspect;
+			}
+		}
+
+		try
+		{
+			if (sender.ShouldSetImageOnAnimationEnd && duration > TimeSpan.Zero)
+			{
+				await Task.Delay(duration, token).ConfigureAwait(ConfigureAwaitOptions.ForceYielding);
```

Comment:
@brminnick I believe you added the `ForceYielding` here, what would be the case? I tried to find the reason to yield it but I failed

Reply:
It was a part of troubleshooting why animations weren't triggering. I now defer calling `ConfigureAwaitOptions.ForceYielding` until `RunAnimationTask()`;